webapp/src/blocks.tsx

```diff
@@ -336,4 +338,13 @@ export class Editor extends srceditor.Editor {
             <sui.Button text={lf("JavaScript") } textClass="ui landscape only" icon="keyboard" onClick={() => this.openTypeScript() } />
         )
     }
+
+    cleanUpShadowBlocks() {
+        const blocks = this.editor.getTopBlocks(false);
+        for (const block of blocks) {
```

Comment:
We tend to use a more functional approach to arrays around the code base:

    blocks.filter(b => b.isShadow_).forEach(b => b.dispose(false))